lib/providers/auth_provider.dart

```diff
@@ -32,6 +34,21 @@ class AuthProvider extends ChangeNotifier {
     _registeredInStatus = value;
   }
 
+  Token? get loginToken => _loginToken;
+
+  Future<void> checkLoginStatus() async {
+    // try to retrieve token if existed
+    final storedToken = await FlutterSecureStorage().read(key: 'login_token');
+    print('stored token_read:$storedToken');
+    if (storedToken != null) {
+      _loginToken = Token.fromJson(json.decode(storedToken));
+      _loggedInStatus = Status.LoggedIn;
+    } else {
+      _loggedInStatus = Status.NotLoggedIn;
+    }
+    notifyListeners();
+  }
+
```

Comment:
Similar to what I was talking to Jason about for the 519 project. We want to verify a token when it's retrieved from the storage. I think this library will help with that. Simply verify that it has an expiration date greater than the current date (be careful with timezones) 

https://pub.dev/packages/jwt_decoder